Remove validations

ladinesa opened the pull request.

ladinesa commented:
@JFRudzinski so I want to retain chemas physical property structure, I like it tbh. I simply removed the unnecessary validations. I tried it on the mapping parser and it seems to work just with the usual m_from_dict.

JFRudzinski commented:
Let's discuss this at today's meeting. The following questions already come to mind: 
Are you proposing to
- apply this to all properties (even simple properties that don't require additional metadata)?
- also keep the flexible structure and variables functionality?
- treat different types within the same repeating subsection under Outputs?
- treat decomposable types with a contributions subsection?

ladinesa committed:
Outputs sub class

ladinesa committed:
Remove setting of value shape

ndaelman-hu commented:
> @JFRudzinski so I want to retain chemas physical property structure, I like it tbh. I simply removed the unnecessary validations. I tried it on the mapping parser and it seems to work just with the usual m_from_dict.

@ladinesa If this passes testing, which aspects are blocking this PR? Is just the git pipeline and my feedback?

ndaelman-hu committed:
- Solve failing pipeline: apply `ElectronicStructureOutputs` where needed

ndaelman-hu committed:
Apply ruff

ndaelman-hu commented:
@ladinesa You mentioned that this setup worked for your new parser. Is this still the case?
I think that extracting `value` should yield an error for any arrays now with the extra typing restrictions.

ndaelman-hu committed:
Remove checks for `shape`, as they are already satisfied by `Quantity` upon definition of `value`

ndaelman-hu committed:
- Remove `test_setattr_none` since `None` should not be blocked from setting
- TODO: clean `PhysicalProperty.value == None` via normalization

ndaelman-hu committed:
- Remove `rank` from `PhysicalProperty`
- Relax shape checking in some cases (`ElectronicBandGap`, `BaseElectronicEigenvalues`, `SpectralProfile`, `QuasiparticleWeight`)
- Fix testing
-- Relax guard for `None` assignments
-- Adapt logic to accommodate generic `shape`

ndaelman-hu committed:
Apply ruff

ndaelman-hu committed:
Fix mypy`

ndaelman-hu committed:
- Scrub out `rank`
- Remove reminders for migrating `iri` and `rank` to `m_def`

ndaelman-hu committed:
Further scrubbing of `rank`

ndaelman-hu committed:
Remove excess properties from `PhyiscalPropert`:
- previously commented out normalization + setter
- new_value
- redundant initializer logic
- redundant normalize logic

ndaelman-hu commented:
Honestly, I don't see the point in `PhysicalProperty.source`.

@JFRudzinski @ladinesa

JFRudzinski commented:
> Honestly, I don't see the point in `PhysicalProperty.source`.
> 
> @JFRudzinski @ladinesa

Can you elaborate on your thoughts? I guess the point is that our PP is living under Simulation() anyway, so the context is assumed, and something like `source` was probably intended for the more general PP class, i.e., when a simulation property gets lifted to results where it is placed in an equivalent section/quantity as the corresponding experimental data?

ndaelman-hu committed:
Remove `PhysicalProperty.variables` in lieu of more specific types

ndaelman-hu committed:
Apply ruff

ndaelman-hu committed:
Solve mypy error

ndaelman-hu committed:
Remove `no_shape_check` where possible

ndaelman-hu commented:
@ladinesa This is ready for testing now with a parser. Note that you may have to modify the shape of a few cases. `PhysicalProperty.variables` is no longer valid, btw. Where necessary, there will be semantically denoted subsections.

I also enforce shape checking as much as possible. There are 2 instances where the shape may really become more nebulous, maybe I can pin them down in the near future.

ndaelman-hu commented:
@JFRudzinski Barring any comments from Alvin, I'd say that this is ready for merging.
There is still further cleaning possible, and we should decide on what we do with `variables.py`, but this already fixes most issues + allows Alvin to continue with the parsers.

ladinesa commented:
> @ladinesa This is ready for testing now with a parser. Note that you may have to modify the shape of a few cases. `PhysicalProperty.variables` is no longer valid, btw. Where necessary, there will be semantically denoted subsections.
> 
> I also enforce shape checking as much as possible. There are 2 instances where the shape may really become more nebulous, maybe I can pin them down in the near future.

i will test this over the weekend.

ladinesa commented on src/nomad_simulations/schema_packages/outputs.py:
```diff
@@ -66,6 +66,8 @@ class Outputs(ArchiveSection):
     )
 
+
+class ElectronicStructureOutputs(Outputs):
```
Maybe redo this change, i remember we agreed that we, for the moment simply put all properties in Outputs

JFRudzinski replied:
@ladinesa I made this change, please check #175 

I also tested this branch with my h5md parser (both mapping annotations and the old way), and had no issues. I suggest that I merge #175 here and then merge this one. Please confirm.

ladinesa replied:
ok, i am also adapting the changes to the parsers.

ladinesa committed:
Add forces shape

JFRudzinski committed:
reverted electronicoutputs (#175)

* reverted electronicoutputs

* reverted from ElectronicStructureOutputs to Outputs in tests

* ruff

---------

Co-authored-by: jrudz <rudzinski@mpip-mainz.mpg.de>

JFRudzinski commented:
@ladinesa Can you have a look at why these tests are failing? Everything works for me locally, and I did not change any dependencies or anything in the toml when I merged my branch. Also, my branch was building correctly on its own....so I have no idea why everything fails now :grimacing:

ladinesa commented:
> @ladinesa Can you have a look at why these tests are failing? Everything works for me locally, and I did not change any dependencies or anything in the toml when I merged my branch. Also, my branch was building correctly on its own....so I have no idea why everything fails now 😬

i am not sure, but maybe use another nomad-lab version 1.3.14 or latest 1.3.16.dev100

JFRudzinski committed:
loosen nomad-lab requirement

JFRudzinski committed:
hard nomad-lab requirement

JFRudzinski committed:
new nomad-lab requirement

JFRudzinski commented:
tried:

nomad-lab>=3.14.0
nomad-lab==3.14.0
nomad-lab>=1.3.16.dev100

will post on discord issues

JFRudzinski committed:
updated wheel constraints

JFRudzinski committed:
relo uv constraint

JFRudzinski commented:
> @ladinesa Can you have a look at why these tests are failing? Everything works for me locally, and I did not change any dependencies or anything in the toml when I merged my branch. Also, my branch was building correctly on its own....so I have no idea why everything fails now 😬

Fix is here on [Discord](https://discordapp.com/channels/1201445470485106719/1356611349685211137)

JFRudzinski commented:
closes #143
